pallet/src/lib.rs

```diff
@@ -158,11 +155,43 @@ pub mod pallet {
             <Authorities<T>>::put(authorities);
         }
 
-        pub fn next_session_authorities() -> Result<Vec<T::AuthorityId>, AlephApiError> {
+        pub fn next_session_authorities() -> Vec<T::AuthorityId> {
             Session::<T>::queued_keys()
                 .iter()
-                .map(|(_, key)| key.get(T::AuthorityId::ID).ok_or(AlephApiError::DecodeKey))
-                .collect::<Result<Vec<T::AuthorityId>, AlephApiError>>()
+                .map(|(_, key)| {
+                    key.get(T::AuthorityId::ID)
+                        .expect("Keys in storage decode correctly.")
```

Comment:
Do we want to potentially panic in the `pallet` code?

Reply:
1) This cannot quite panic, the only way for this to panic is to use a wrong `AuthoridyId::ID`, which you can catch during the first compilation. Also, why would panicking in the runtime be worse than panicking in the client code? In the previous version the same panic was just one level up...
2) This makes the output type of the calls
`next_session_validator_keys`
`current_session_validator_keys`
identical.

Reply:
Sure I agree. I am only asking this because the substrate explicitly says the runtime should handle errors and not panic: `https://substrate.dev/docs/en/knowledgebase/runtime/errors`. I know there are more places like this in code so I wanted to know what is our stand on this topic.

Reply:
The substrate doc seems pretty adamant about this, so I would support changing this to an `Err`.